Sources/Sentry/SentryScope.m

```diff
@@ -59,6 +59,8 @@ @implementation SentryScope {
     NSObject *_spanLock;
 }
 
+@synthesize span = _span;
```

Comment:
m: Actually, we don’t need any of the changes about span in the `.m` file, just need to change the span property declaration from `nonatomic` to `atomic`, same result.

eg: `@property (nullable, atomic, strong) id<SentrySpan> span;`

P.S.: You don’t need this `@synthesize` if you are using the standard name declaration (_ in front of the name) like this one.

Reply:
Not totally true, we could remove the existing `setSpan` function.

Reply:
When I remove the `synthesize` I get:  `Use of undeclared identifier '_span'; did you mean 'span'?` I'm a bit confused.

`atomic` is a good point, but I would like to include the scope observer call in the same lock as setting the span to ensure the observers and the scope are in sync. That doesn't work correctly when using atomic.
```objc
@synchronized(_spanLock) {
    _span = span;

    for (id<SentryScopeObserver> observer in self.observers) {
        [observer setTraceContext:[self buildTraceContext:span]];
    }
}
```

Reply:
> When I remove the synthesize I get:  Use of undeclared identifier '_span'; did you mean 'span'? I'm a bit confused.

I mean the entire line.
```@synthesize span = _span;```

> but I would like to include the scope observer call in the same lock as setting the span

True. Thanks!!

Reply:
> I mean the entire line.
> @synthesize span = _span;

Yes, I removed the entire line.